packages/mantine/src/components/modal/Modal.tsx

```diff
@@ -0,0 +1,33 @@
+import {Modal as MantineModal, ModalProps, Text} from '@mantine/core';
+import {FunctionComponent} from 'react';
+import {Header, HeaderProps as PlasmantineHeaderProps} from '../header';
+
+interface HeaderProps extends Omit<PlasmantineHeaderProps, 'children'> {
+    title: string;
+}
+
+export interface CustomModalProps extends Omit<ModalProps, 'centered' | 'title'> {
+    /**
+     * Header component props (layout/Header)
+     */
+    header: HeaderProps;
+}
+
+export const Modal: FunctionComponent<CustomModalProps> = ({
```

Comment:
Don't forget to add api styles and theming support https://mantine.dev/guides/custom-components/